internal/command/session.go

```diff
@@ -400,7 +400,19 @@ func (c *Commands) checkSessionTerminationPermission(ctx context.Context, model
 	if model.UserID != "" && model.UserID == authz.GetCtxData(ctx).UserID {
 		return nil
 	}
-	return c.checkPermission(ctx, domain.PermissionSessionDelete, model.UserResourceOwner, model.UserID)
+	// Before 2.42.0, the resourceOwner of a session used to be the organisation of the creator.
+	// Further the (checked) users organisation id was not stored.
+	// To be able to check the permission, we need to get the user's resourceOwner in this case
+	userResourceOwner := model.UserResourceOwner
+	if model.UserID != "" && userResourceOwner == "" {
+		r := NewResourceOwnerModel(ctx, user.AggregateType, model.UserID)
```

Comment:
Could move it to a separate function t just return the resource owner and the error